src/Metadata/HeaderParameter.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata;
+
+class HeaderParameter extends Parameter implements HeaderParameterInterface
```

Comment:
Would be safer to declare it as `@experimental` in next minor version (e.g.: 3.4.0), and remove the `@experimental` on the after next minor version (e.g.: 3.5.0).
Same for HeaderParameterInterface, QueryParameter, QueryParameterInterface and `parameters` argument on Metadata/*